### ISSUE TITLE
.ltr, .wrd and lexicon files for train and infer letters model. 

### ISSUE THREAD

**Issue by fullmacht** [facebookresearch/fairseq#4308]
## ❓ Questions and Help

Fine-tuning model.
1. For letters model i need to have .ltr file like this example below?
H A V E | A | N I C E |
H E L L O |
2. For the fine-tuning letters model i have to set the comand `+criterion.wer_lexicon=path/to/lexicon.txt`. "lexicon.txt" file have to be like this example below?
H A V E | A | N I C E |
H E L L O |
3. For fine-tuning model with the comand `+criterion.wer_lexicon=path/to/lexicon.txt`.  "lexicon.txt" file must contain only train text?
4. To build KenLM for fine-tuning model i have to set this "lexicon.txt" file in a command. The "lexicon.txt" file must contain only train text or all text from train and dev_other ( validation ) texts?

Evaluation model.
5. For the evaluating letters model i have to set the comand `decoding.lexicon=/path/to/lexicon.txt`. "lexicon.txt" file must contain only dev_other  ( validation ) text?
6. Ken_LM model need to be builded with "lexicon.txt" file that contains only dev_other ( validation text )?

I fine-tuned letter model with ~500 hours dataset 17 epoch with `--config-name base_960h` and got this train data:

[train_inner][INFO] - {"epoch": 17, "update": 16.104, "loss": "50.711", "ntokens": "3579.69", "nsentences": "128", "nll_loss": "1.813", "wps": "1087.3", "ups": "0.3", "wpb": "3579.7", "bsz": "128", "num_updates": "44000", "lr": "0.0001", "gnorm": "48.35", "loss_scale": "64", "train_wall": "3426", "gb_free": "9.3", "wall": "60378"}

and got this validation data:

[dev_other][INFO] - {"epoch": 17, "dev_other_loss": "33.851", "dev_other_ntokens": "223.162", "dev_other_nsentences": "7.99905", "dev_other_nll_loss": "1.213", "dev_other_uer": "25.292", "dev_other_wer": "98.469", "dev_other_raw_wer": "57.074", "dev_other_wps": "1099.9", "dev_other_wpb": "223.2", "dev_other_bsz": "8", "dev_other_num_updates": "44000", "dev_other_best_wer": "98.469"}

At the 2-nd epoch the train result was so:
![image](https://user-images.githubusercontent.com/45352024/160095056-822a5495-f28f-42d1-923c-83bae0a9b05e.png)

At the 3-d epoch the test result was so:
![image](https://user-images.githubusercontent.com/45352024/160095357-3847fd0e-c4b6-4748-8301-f2fd9374f313.png)

#### What's your environment?

 - fairseq Version: '1.0.0a0+b554f5e'
 - PyTorch Version: 1.10.0a0+git302ee7b
 - OS Linux
 - Python version: 3.8
 - GPU models and configuration: 3060 12GB
 - Any other relevant information:
